addons/account/models/account_move.py

```diff
@@ -1013,14 +1013,40 @@ def _compute_fiscal_position_id(self):
             move.fiscal_position_id = self.env['account.fiscal.position'].with_company(move.company_id)._get_fiscal_position(
                 move.partner_id, delivery=delivery_partner)
 
-    @api.depends('bank_partner_id')
+    @api.depends('bank_partner_id', 'currency_id', 'preferred_payment_method_line_id')
     def _compute_partner_bank_id(self):
         for move in self:
-            # This will get the bank account from the partner in an order with the trusted first
-            bank_ids = move.bank_partner_id.bank_ids.filtered(
+            if (
+                payment_method := (
+                    move.preferred_payment_method_line_id
+                    or (
+                        move.bank_partner_id.property_inbound_payment_method_line_id
+                        if move.is_inbound()
+                        else move.bank_partner_id.property_outbound_payment_method_line_id
+                    )
+                )
+            ) and payment_method.journal_id:
+                move.partner_bank_id = payment_method.journal_id.bank_account_id
+                continue
+
+            def _bank_selection_key(bank):
```

Comment:
Nice! that's exactly what I was thinking.
Though this method doesn't need to be defined in the loop...